instant.ts

```diff
@@ -78,23 +78,48 @@ const orderPackageIds = (allPackages, chosenPackageIds) => {
   const orderedPackageIds = []
   const packagesWithDependencies = []
 
-  chosenPackageIds.forEach(id => {
-    if (
-      !allPackages[id].metadata.dependencies ||
-      !allPackages[id].metadata.dependencies.length
-    ) {
-      orderedPackageIds.push(id)
-    } else {
-      allPackages[id].metadata.dependencies.forEach(dependency => {
-        if (!Object.keys(allPackages).includes(dependency)) {
-          throw Error(`Dependency ${dependency} for package ${id} does not exist`)
+  function collectPackagesWithDependencies(chosenIds) {
+    let inexplicitDependencies = []
+
+    chosenIds.forEach(id => {
+      if (id && allPackages[id] && allPackages[id].metadata) {
+        if (
+          !allPackages[id].metadata.dependencies ||
+          !allPackages[id].metadata.dependencies.length
+        ) {
+          if (id === 'core') {
+            orderedPackageIds.unshift(id)
+          } else {
+            orderedPackageIds.push(id)
+          }
+        } else {
+          allPackages[id].metadata.dependencies.forEach(dependency => {
+            if (!Object.keys(allPackages).includes(dependency)) {
+              throw Error(`Dependency ${dependency} for package ${id} does not exist`)
+            }
+            if (!chosenIds.includes(dependency)) {
+              inexplicitDependencies.push(dependency)
+            }
+          })
+          packagesWithDependencies.push(id)
         }
-      })
-      packagesWithDependencies.push(id)
+      } else {
+        throw Error(`Package ${id} does not exist or the metadata is invalid`)
+      }
+    })
+
+    if (inexplicitDependencies.length) {
+      collectPackagesWithDependencies(inexplicitDependencies)
```

Comment:
This recursive function needs protection against circular dependencies else it looks until `RangeError: Maximum call stack size exceeded`

Reply:
@rcrichton addressed in https://github.com/openhie/instant/pull/63/commits/5a7840d5f0a619966b6639e6705598c7ff35dae6